modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -1920,15 +1922,26 @@ public ObjectEntry updateStatus(
 			ServiceContext serviceContext)
 		throws PortalException {
 
-		if (objectEntry.getStatus() == status) {
+		if ((objectEntry.getDisplayDate() == null) &&
```

Comment:
double check this condition according to the AC, please

Reply:
The order doesn't affect the logic since the "&&" operator is commutative. The AC3 still being covered.

Reply:
I reviewed, and this condition will not impact scenarios prior to this story implementation. 

According to @JhossephG, this condition was added because there is a scenario where the `objectEntry.getStatus() == status` but the `objectEntry.getDisplayDate() != null`, and this method's logic needs to be performed.